### ISSUE TITLE
AArch64 Supported?

### ISSUE THREAD

**Issue by eusru** [CesiumGS/cesium-unreal#897]
Linux AArch64 Supported ？
After I package for and run on Ubuntu AArch64，Got Errors about “can not find files Cesium “
Most of the Errors are “Cannot find cesium”
I guess Cesium load failed on AArch64

**Comment by kring** [comment]
We've never tried it, but I can't think of any reason it won't work. Did you follow the Developer Setup guide? It sounds like you might simply be missing the built cesium-native libs.
https://github.com/CesiumGS/cesium-unreal/blob/ue5-main/Documentation/developer-setup.md

It's also possible some small tweaks will be needed to the build scripts on either the cesium-native or the cesium-unreal side.

If you've followed the guide and are still having trouble, please post your complete log.

**Comment by eusru** [comment]
> We've never tried it, but I can't think of any reason it won't work. Did you follow the Developer Setup guide? It sounds like you might simply be missing the built cesium-native libs. https://github.com/CesiumGS/cesium-unreal/blob/ue5-main/Documentation/developer-setup.md
> 
> It's also possible some small tweaks will be needed to the build scripts on either the cesium-native or the cesium-unreal side.
> 
> If you've followed the guide and are still having trouble, please post your complete log.

Emmmm.I just downloaded from Unreal Marketplace . Is there any differences between UnrealMarketplace  and GitHub SourceCode

**Comment by kring** [comment]
Yes, the Unreal Marketplace has pre-built cesium-native libraries for the supported platforms, which does not include Linux/AArch64. To build for AArch64, you'll need to build those libraries from source, and the easiest way to do that is by following the developer setup guide.

**Comment by kring** [comment]
Well, I can see from the log that the immediate problem is that `CMAKE_C_COMPILER` is somehow set to `aarch64-linux-gnu-gcc`, which doesn't make any sense. How that's happening is less clear.

You probably need to copy the triplet we use for Linux and x86-64, x64-linux-unreal.cmake, and modify it for arm64. I don't think `VCPKG_C_COMPILER` is actually a valid variable to set it a triplet file. You need to set the compiler via a CMake toolchain instead, which is what our `x64-linux-unreal.cmake` does.

**Comment by Tommy-T1984** [comment]
> Well, I can see from the log that the immediate problem is that `CMAKE_C_COMPILER` is somehow set to `aarch64-linux-gnu-gcc`, which doesn't make any sense. How that's happening is less clear.
> 
> You probably need to copy the triplet we use for Linux and x86-64, `x64-linux-unreal.cmake`, and modify it for arm64. I don't think `VCPKG_C_COMPILER` is actually a valid variable to set in a triplet file. You need to set the compiler via a CMake toolchain instead, which is what our `x64-linux-unreal.cmake` does.

Do you compile it, successfully?

**Comment by YooRan-yty** [comment]
On a LinuxArm64 system machine, I cloned the Cesium-Unreal source code with

`git clone --recursive https://github.com/CesiumGS/cesium-unreal.git`



Following the guide documentation, I successfully compiled the Cesium-Unreal runtime libraries with
`cmake --build build --target install`

After that, a folder appeared in `CesiumForUnreal/Source/ThirdParty/lib` named `Linux-arm-Release`, which contains many `*.a` files and a `pkgconfig` directory.
I think this means I succeeded.

I copied this folder into my plugin on Windows and used the UE engine to package my project for the LinuxArm64 platform.
The packaging went smoothly (previously, before I compiled these runtime libraries, I couldn’t successfully package for LinuxArm64).

However, when I ran the packaged project on the LinuxArm64 platform, the application itself launched, but the Cesium terrain failed to load properly.
I checked the log file and found multiple `LogStream::Error` messages related to Cesium, roughly indicating `“Missing Dependency”`.

What could be the reason for this? Is this an issue with the packaging process, or is it a problem with the libraries (since I didn’t perform the step of compiling the plugin on Linux, with the`./RunUAT......`)?
I’d appreciate your reply. Thank you.

**Comment by kring** [comment]
@YooRan-yty we'll probably need to see the actual error messages to have any guess about what might have gone wrong.

**Comment by YooRan-yty** [comment]
Thanks, I switched the Unreal Engine version. I used 5.5.4 to package my project, and now my Cesium is working correctly on my Arm Linux machine!

**Comment by zhangzekai725-art** [comment]
Hi,I'm facing a big problem on my ue5.4 project with Cesium on my Windows.And I'll be very appreciate if you could give me some Detailed advice since I'm not so experienced.Now,I want to package my project to "Linux Arm64",but it failed,and the log said" Unable to instantiate module CesiumRuntime: System nvalidOperationException Cesium for Urreal does not support this platform." I wonder if there ia any way to package it as easily as possible?I had succeed to packaged it to "Linux" before ,it only took me some hours.By the way ,please tell me how long will I spend if the"Linux Arm64" works.Thanks!!

<img width="1478" height="879" alt="Image" src="https://github.com/user-attachments/assets/fb96f6de-9abe-4fa5-883b-97e38d2d2db3" />

**Comment by yuexinlp** [comment]
> 谢谢，我已经更换了虚幻引擎版本。我用的是 5.5.4 版本打包我的项目，现在我的 Cesium 在我的 Arm Linux 机器上运行正常了！
Must compilation be performed on a Linux ARM machine?
